### PR TITLE
Update storiesof api docs

### PR DESCRIPTION
**Pull request by neilyio**
Issue:
The documentation for the `storiesOf` API is missing some important points, as laid out in #12288 

## What I did
- mention of `.addDecorators` and `.addParameters` for component-level story configuration
- quick explainer of global-level parameters with links back to the main documentation.
- explanation of second argument to `storiesOf`, the `module` object.

## How to test
Take a quick look, I did my best!

### CODE REVIEW

File: lib/core/docs/storiesOf.md [hunk 1]
```diff
@@ -57,6 +59,27 @@ storiesOf('Button', module).add(
...
+        😀 😎 👍 💯
+      </span>
+    </Button>
+  ));
+```
+
+Finally, global parameters for all stories can be set via the parameters export of your `.storybook/previews.js`, see [this page](https://storybook.js.org/docs/react/writing-stories/parameters#global-parameters) for an example.
+
+Global decorators for all stories can be set via the decorators export of your `.storybook/preview.js`, see [this page](https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) for an example.
+
```
Comment (jonniebigodes) [inline lib/core/docs/storiesOf.md:82]: @neilyio i think we could do better and probably go with something like:

 " Parameters and decorators can also be used globally, you can define them in your `.storybook/preview.js`. Take a look [here](https://storybook.js.org/docs/react/writing-stories/parameters#global-parameters) to learn more about global parameters and [here](https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) for global decorators."

Let me know if you're ok with it

File: lib/core/docs/storiesOf.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ storiesOf('Button', module)
 
 The string argument to `storiesOf` is the component title. If you pass a string like `'Widgets|Button/Button'` it can also be used to position your component's story within Storybook's story hierarchy.
 
+The second argument to `storiesOf` is the module object, the one you might be familiar with from ES5's `module.exports`. Storybook needs a reference to the file/module where your story lives to enable hot-module-replacement. If you do not supply it, you'd need to refresh your browser for every change you make.
+
```
Comment (jonniebigodes) [inline lib/core/docs/storiesOf.md:35]: @neilyio we have a bit of room for improvement here. We could probably go with:

" The second argument of `storiesOf` is a reference to each story available. Storybook needs it to enable hot-module-replacement. If it's not included you'll need to refresh your browser with each change you make."

Let me know if you're ok with it.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@neilyio @shilman @tmeasday i was checking this pull request and proposed some changes. 

Hope you're all ok with them.

Feel free to provide feedback

**Comment by neilyio** [comment]
@jonniebigodes good feedback, just made your changes.

**Comment by jonniebigodes** [comment]
wow... @neilyio glad that you were ok with the proposed changes and managed to address them so soon. 

If both @tmeasday and @shilman are ok with it this will get merged.

Do you mind waiting just a bit longer while they take a look at it?

Stay safe

**Comment by neilyio** [comment]
For sure! Just glad we can address this. This is my first time contributing to anything, so it's kind of exciting! Wasn't even sure how to fork/pull request before.

There are some more additions that need to be made to this file, we're discussing them in issue #12302. It needs to be clear to the user how to use `args` and `argTypes` with `storiesOf`. I'm not sure how they're supposed to work myself, so I'm refraining from adding to the docs just yet.

**Review by shilman — APPROVED** [review]
Looks great @neilyio 💯

File: lib/core/docs/storiesOf.md [hunk 3]
```diff
@@ -31,6 +31,8 @@ storiesOf('Button', module)
 
 The string argument to `storiesOf` is the component title. If you pass a string like `'Widgets|Button/Button'` it can also be used to position your component's story within Storybook's story hierarchy.
 
+The second argument of `storiesOf` is a reference to each story available. Storybook needs it to enable hot-module-replacement. If it's not included you'll need to refresh your browser with each change you make.
```
Comment (tmeasday) [inline lib/core/docs/storiesOf.md:34]:
```suggestion
The second argument of `storiesOf` is a webpack `module`, which is available on the global (per-file) scope. Storybook needs it to enable hot-module-replacement. If it's not included you'll need to refresh your browser with each change you make.
```

File: lib/core/docs/storiesOf.md [hunk 4]
```diff
@@ -57,6 +59,25 @@ storiesOf('Button', module).add(
 );
 ```
 
+We can control how the component's stories will render with parameters and decorators. You can use as many `.addDecorators` as you need, but you can only use one `.addParameters`, as you can see in the example below:
```
Comment (tmeasday) [inline lib/core/docs/storiesOf.md:62]:
```suggestion
We can control how the component's stories will render with parameters and decorators. You can use as many `.addDecorators` as you need (but make sure you add them all before your first story), but you can only use one `.addParameters`, as you can see in the example below:
```

**Review by tmeasday — APPROVED** [review]
LGTM, couple of tweaks.

**Comment by jonniebigodes** [comment]
@neilyio @tmeasday sorry for not getting back to this, but i had to shift my attention elsewhere. I wanted to know what's the status of this? Is it good to go, or will it need some additional attention?

Let me know 

Stay safe.

**Comment by tmeasday** [comment]
Ready to go from my perspective.

**Comment by jonniebigodes** [comment]
@neilyio going to merge this, should any additional work needs to be implemented we can do a follow up pull request and address it.

Sounds good?

**Comment by neilyio** [comment]
Sounds great, thanks for looking it over!

**Comment by jonniebigodes** [comment]
@neilyio no problem and thank you for making the contribution.

**Pull request merged**